src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -98,11 +101,16 @@ public function __construct(HttpKernelInterface $kernel, StoreInterface $store,
             'stale_if_error' => 60,
             'trace_level' => 'none',
             'trace_header' => 'X-Symfony-Cache',
+            'terminate_on_cache_hit' => true,
         ], $options);
 
         if (!isset($options['trace_level'])) {
             $this->options['trace_level'] = $this->options['debug'] ? 'full' : 'none';
         }
+
+        if (true === $this->options['terminate_on_cache_hit']) {
+            trigger_deprecation('symfony/http-kernel', '4.4', 'Setting "terminate_on_cache_hit" to "true" is deprecated and will be changed to "false" in Symfony 7.0.');
```

Comment:
i am no maintainer, but i think it only makes sense to deprecate this for the upcoming symfony 6.2

Reply:
Indeed, we cannot add a new deprecation in a stable branch, only in 6.2 as of today.

Reply:
To reduce burden for developers not using the terminate event, the depreciation message could be triggered only when the event have dispatcher for this event. This code should be in the terminate method of this class.